src/analyzer/analyzermanager.cpp

```diff
@@ -0,0 +1,295 @@
+                //ensure i won't call pause twice for the same worker.
+                m_pausedWorkers.append(backwork);
+                m_backgroundWorkers.removeAll(backwork);
+        }
+        while (m_foregroundWorkers.size() > threads) {
+                AnalyzerWorker * backwork = m_foregroundWorkers.first();
+                backwork->pause();
+                //Ideally i would have done this on the slotPaused slot, but then i cannot 
+                //ensure i won't call pause twice for the same worker.
+                m_pausedWorkers.append(backwork);
+                m_foregroundWorkers.removeAll(backwork);
+        }
+    }
+    else {
+        //resume workers
+        int pendingworkers=threads-m_MaxThreads;
+        foreach(AnalyzerWorker* worker, m_pausedWorkers) {
+            if (!worker->isBatch() && pendingworkers > 0) {
+                worker->resume();
+                m_pausedWorkers.removeOne(worker);
+                m_foregroundWorkers.append(worker);
+                --pendingworkers;
+            }
+        }
+        foreach(AnalyzerWorker* worker, m_pausedWorkers) {
+            if (worker->isBatch() && pendingworkers > 0) {
+                worker->resume();
+                m_pausedWorkers.removeOne(worker);
+                m_backgroundWorkers.append(worker);
+                --pendingworkers;
+            }
+        }
+        //Create new workers, if tracks in queue.
+        pendingworkers = math_min(pendingworkers,m_batchTrackQueue.size());
+        for ( ;pendingworkers > 0; --pendingworkers) {
+            createNewWorker(true);
+        }
+    }
+    m_MaxThreads=threads;
+}
+
+AnalyzerWorker* AnalyzerManager::createNewWorker(bool batchJob) {
+    QThread* thread = new QThread();
+    AnalyzerWorker* worker = new AnalyzerWorker(m_pConfig, ++m_nextWorkerId, batchJob);
+    worker->moveToThread(thread);
+    //Auto startup and auto cleanup of worker and thread.
+    connect(thread, SIGNAL(started()), worker, SLOT(slotProcess()));
+    connect(worker, SIGNAL(finished()), thread, SLOT(quit()));
+    connect(worker, SIGNAL(finished()), worker, SLOT(deleteLater()));
+    connect(thread, SIGNAL(finished()), thread, SLOT(deleteLater()));
+    //Connect with manager.
+    connect(worker, SIGNAL(updateProgress(int, struct AnalyzerWorker::progress_info*)), this, SLOT(slotUpdateProgress(int, struct AnalyzerWorker::progress_info*)));
+    connect(worker, SIGNAL(waitingForNextTrack(AnalyzerWorker*)), this, SLOT(slotNextTrack(AnalyzerWorker*)));
+    connect(worker, SIGNAL(paused(AnalyzerWorker*)), this, SLOT(slotPaused(AnalyzerWorker*)));
+    connect(worker, SIGNAL(workerFinished(AnalyzerWorker*)), this, SLOT(slotWorkerFinished(AnalyzerWorker*)));
+    connect(worker, SIGNAL(error(QString)), this, SLOT(slotErrorString(QString)));
+    thread->start(QThread::LowPriority);
+    if (batchJob) {
+        m_backgroundWorkers.append(worker);
+    }
+    else {
+        m_foregroundWorkers.append(worker);
```

Comment:
It's a bad naming from my side, but I couldn't think of a better one at that time.
Strictly speaking, everything is run in background (i.e. the user does not need to wait for any of them to stop). 
background, in this case, means that they are used for the job of the analysis queue (analysis feature of the library), while foreground means that they run an analysis by an immediate action of the user (just loaded a track).

Maybe I should have used worker and priorityworker. Take in mind that the priority is not the only difference, since there are some Mixxx settings that affect them in a different way. (namely the waveform creation and the bpm analysis)